### PR TITLE
ci: add dead code analysis

### PR DESCRIPTION
**Pull request by tych0**
Not clear if this is really that useful, since it didn't catch a ton of
actually useful stuff. But I did it, so let's see. Using confidence < 100
is more interesting (patches to come), but it also has a ton of false
positives.

Signed-off-by: Tycho Andersen <tycho@tycho.pizza>

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -74,9 +74,15 @@ commands =
 deps = -r{toxinidir}/docs/requirements.txt
 commands = python3 setup.py build_sphinx -W
 
+[testenv:vulture]
+deps =
+    vulture
+commands =
+    vulture --min-confidence=100 --exclude test/configs/syntaxerr.py libqtile test
```
Comment (m-col) [inline tox.ini:81]: So this will fail tests only when it finds something it is 100% sure about?

**Comment by m-col** [comment]
What does prepending the underscores to kwargs do?

**Review by ramnes — APPROVED** [review]

File: tox.ini (hunk 1 above)
Comment (tych0) [reply tox.ini:81]: Yes, exactly. Unfortunately due to the nature of python, that is not very much stuff.

**Comment by tych0** [comment]
> What does prepending the underscores to kwargs do?

It's a convention to say "this is an unused parameter", which causes vulture to ignore the fact that it's unused. This IMO is better than `#noqa` comments where we can't delete args (e.g. test fixtures or because they're arguments defined by a superclass or whatever). I did have to use `noqa` in a couple of places, because changing the names of arguments results in failures due to pytest's crazy introspection :(

**Pull request merged**